### PR TITLE
Make nonroot default runtime

### PR DESCRIPTION
**Pull request by xpivarc**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Kubevirt now runs with nonroot user by default
```

### CODE REVIEW

**Comment by xpivarc** [comment]
/test pull-kubevirt-e2e-k8s-1.22-sig-compute-realtime
Expected failure

File: pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go [hunk 1]
```diff
@@ -1468,8 +1468,18 @@ func validateHostNameNotConformingToDNSLabelRules(field *k8sfield.Path, spec *v1
 	return causes
 }
 
-func validateRealtime(field *k8sfield.Path, spec *v1.VirtualMachineInstanceSpec) (causes []metav1.StatusCause) {
+func validateRealtime(field *k8sfield.Path, spec *v1.VirtualMachineInstanceSpec, nonroot bool) (causes []metav1.StatusCause) {
 	if spec.Domain.CPU != nil && spec.Domain.CPU.Realtime != nil {
+		if nonroot {
```
Comment (xpivarc) [inline pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go:1473]: @vladikr Maybe adding the sys_nice for time being is better option

Comment (vladikr) [reply pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go:1473]: I think we should add sys_nice back for the time being - I agree

**Comment by xpivarc** [comment]
@acardace PTAL

**Comment by rmohr** [comment]
Hi Lubo. Great change. Does this mean that we test now in the `nonroot` lanes the same as in the other CI lanes?

**Comment by xpivarc** [comment]
> Hi Lubo. Great change. Does this mean that we test now in the `nonroot` lanes the same as in the other CI lanes? And as such not testing root anymore?

For a brief time yes, https://github.com/kubevirt/project-infra/pull/2418 should fix it. We should talk about how long we want to do this in the next community meeting.

**Pull request merged**